js/sign/src/utils/utils.ts

```diff
@@ -31,15 +36,63 @@ export function parsePemKey(
   });
 }
 
-export function getRawPublicKey(publicKey: crypto.KeyObject) {
-  // Currently this is the only way for us to get the raw 32 bytes of the public key.
-  return new Uint8Array(
-    publicKey.export({ type: 'spki', format: 'der' }).slice(-32)
+export function isAsymmetricKeyTypeSupported(key: crypto.KeyObject) {
+  return (
+    key.asymmetricKeyType === 'ed25519' ||
+    (key.asymmetricKeyType === 'ec' &&
+      key.asymmetricKeyDetails!.namedCurve === 'secp256k1')
+  );
+}
+
+export enum SignatureType {
+  Ed25519,
+  EcdsaP256SHA256,
+}
+
+export function getSignatureType(key: crypto.KeyObject) {
+  assert(
+    isAsymmetricKeyTypeSupported(key),
+    'Expected either "Ed25519" or "ECDSA P-256" key.'
   );
+  if (key.asymmetricKeyType === 'ed25519') {
+    return SignatureType.Ed25519;
+  }
+  return SignatureType.EcdsaP256SHA256;
+}
+
+export function getPublicKeyAttributeName(key: crypto.KeyObject) {
+  switch (getSignatureType(key)) {
+    case SignatureType.Ed25519:
+      return ED25519_PK_SIGNATURE_ATTRIBUTE_NAME;
+    case SignatureType.EcdsaP256SHA256:
+      return ECDSA_P256_SHA256_PK_SIGNATURE_ATTRIBUTE_NAME;
```

Comment:
With the enum+map combo i believe you could even do something like [this](https://stackoverflow.com/a/39372911), but maybe it's not so cool with just 2 supported key types. Are we expecting to have more key types supported?

Reply:
Done!